lib/engine.py

```diff
@@ -264,32 +260,38 @@ def auto_get_delimiter(self, header):
 
     def convert_data_type(self, datatype):
         """Converts Retriever generic data types to database platform specific
-        data types"""
-        thistype = datatype[0]
+        data types
+        """
+        # get the type from the dataset variables
+        key = datatype[0]
         thispk = False
-        if thistype[0:3] == "pk-":
-            thistype = thistype.lstrip("pk-")
+        if key[0:3] == "pk-":
+            key = key.lstrip("pk-")
             thispk = True
-        elif thistype[0:3] == "ct-":
-            thistype = thistype[3:]
-
-        if thistype in list(self.datatypes.keys()):
-            thistype = self.datatypes[thistype]
+        elif key[0:3] == "ct-":
+            key = key[3:]
 
+        # format the dataset type to match engine specific type
+        thistype = ""
```

Comment:
Nevermind, I see why you did it this way. This function is confusing to read (hence my reviewing mistakes), but I agree we should just leave it like it is at the moment since I can't figure out a better way to write it.